tests/integration/security/authn/authn_util.go

```diff
@@ -0,0 +1,60 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package authn
+
+import (
+	"fmt"
+
+	"istio.io/istio/tests/integration/security/util/connection"
+)
+
+type TestCase struct {
+	Request             connection.Checker
+	ExpectAuthenticated bool
+}
+
+func getErrorMessage(req connection.Checker, expect, actual string) error {
+	return fmt.Errorf("%s to %s:%s using %s: expected %s, got %s",
+		req.From.Config().Service, req.Options.Target.Config().Service, req.Options.PortName, req.Options.Scheme, expect, actual)
+}
+
+// CheckAuthn checks a request based on ExpectAuthenticated (true: resp code 200; false: resp code 401 ).
```

Comment:
Comment is not very clear. Do you mean return nil if response code matches with c.Expected?

Reply:
Let me update the comment when cherry-picking to 1.2.

Reply:
Actually, not sure if doing that will make merging back to master harder (e.g conflicts). Basically, if ExpectAuthenticated is true, CheckAuthn expects the response code to be 200, and 401 if ExpectAuthenticated is false. I will update the comment once we merge 1.2 back to master.